core/jazz_environment-event-handler/test/test.js

```diff
@@ -15,642 +15,1496 @@
+        sinon.assert.calledTwice(requestPromiseStub);
+        requestPromiseStub.restore();
+        expect(res[0].message).to.eql('Stage and Prod environments are created successfully');
+      });
+  });
+  it('handleError should return error response json for valid input', function () {
+    let errorJson = { "failure_code": 400, "failure_message": "Unauthorized" };
+    let handleError = index.handleError(400, "Unauthorized");
+    assert(handleError, errorJson);
+  });
+
+  it('handleProcessedEvents should push valid events', function () {
+    let event = { "Records": [testPayloads.eventPayload] }
+    let sequenceNumber = event.Records[0].kinesis.sequenceNumber;
+    let encodedPayload = event.Records[0].kinesis.data;
+    let handleProcessedEvents = index.handleProcessedEvents(encodedPayload, sequenceNumber);
+    expect(index.getEventProcessStatus()).to.have.property('processed_events');
+  });
+
+  it('handleFailedEvents should push valid events', function () {
+    let event = { "Records": [testPayloads.eventPayload] }
+    let sequenceNumber = event.Records[0].kinesis.sequenceNumber;
+    let encodedPayload = event.Records[0].kinesis.data;
+    let handleFailedEvents = index.handleFailedEvents(encodedPayload, sequenceNumber, "101", "failure");
+    expect(index.getEventProcessStatus()).to.have.property('failed_events');
+  });
+
+  it('getEventProcessStatus should return failed/processed records length', function () {
+    expect(index.getEventProcessStatus()).to.have.property('processed_events');
+    expect(index.getEventProcessStatus()).to.have.property('failed_events');
+    expect(index.getEventProcessStatus()).to.not.have.property('some_key');
+  });
+
+  it('handler should fail for invalid authentication', function () {
+    let event = { "Records": [testPayloads.eventPayload] }
+    let responseObject = {
+      statusCode: 400,
+      body: { "message": "unautho" }
+    };
+
+    let authStub = sinon.stub(rp, 'Request').returns(Promise.resolve(responseObject));
+    requestPromiseStub.callsFake((obj) => {
+      return obj.callback(null, responseObject, responseObject.body);
+    });
+
+    index.handler(event, context, (err, res) => {
+      if (err) {
+        return err;
+      } else {
+        res.should.have.property('processed_events');
+        return res;
+      }
+    });
+
+    authStub.restore();
+    requestPromiseStub.restore();
+  });
+
+  it('handler should resolve for not interested events', function () {
+    let event = { "Records": [testPayloads.eventPayload] }
+    let responseObject = {
+      statusCode: 400,
+      body: { "message": "unautho" }
+    };
+
+    let authStub = sinon.stub(rp, 'Request').returns(Promise.resolve(responseObject));
+    requestPromiseStub.callsFake((obj) => {
+      return obj.callback(null, responseObject, responseObject.body);
+    });
+    let message = 'User is not authorized to access this service';
+    index.handler(event, context, (err, res) => {
+      if (err) {
```

Comment:
try not to use if and else in tests. Ensure your test is set up to either get err or data. You test should be deterministic. Split this and make 2 tests. One which surely gives and error and another with data(success). Please update other tests if you have taken similar approach

Reply:
Done